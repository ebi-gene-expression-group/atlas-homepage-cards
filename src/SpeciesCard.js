import React from 'react'
import PropTypes from 'prop-types'
import EbiSpeciesIcon from 'react-ebi-species'

const SpeciesCard = ({iconSrc, description, content}) => {
  return (
    <div className="column column-block text-center combo" style={{marginBottom:0, paddingBottom: `25px`}}>
      <span className="large-species-icon" style={{fontSize: `600%`}}>
        <EbiSpeciesIcon species={iconSrc}/>
      </span>

      <h5 className="species">{description}</h5>
      {
        content.map((item) =>
          item.url ?
            <p style={{marginBottom: `0.3rem`}} key={item.text}><a href={item.url} key={item.text}>{item.text}</a></p> :
            <p style={{marginBottom: `0.3rem`}} key={item.text}>{item.text}</p>
        )
      }
    </div>
  )
}

SpeciesCard.propTypes = {
  iconSrc: PropTypes.string,
  description: PropTypes.string,
  content: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string
  }))
}

export default SpeciesCard
