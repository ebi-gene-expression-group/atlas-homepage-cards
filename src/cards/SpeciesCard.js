import React from 'react'
import PropTypes from 'prop-types'
import EbiSpeciesIcon from 'react-ebi-species'

import cardPropTypes from './CardPropTypes'

const SpeciesCard = ({iconSrc, description, content}) =>
  <div className={`column column-block text-center combo card`} style={{marginBottom:0, paddingBottom: `25px`}}>
    {
      description && description.url ?
        <a className={`species-icon`} style={{fontSize: `600%`, borderBottom: 0}} href={description.url}>
          <EbiSpeciesIcon species={iconSrc}/>
        </a> :
        <span className={`species-icon`} style={{fontSize: `600%`}}>
          <EbiSpeciesIcon species={iconSrc}/>
        </span>
    }

    {
      description &&
      <h5 className={`species-name`}>
        {
          description.url ?
            <a className={`species-url`} href={description.url}>{description.text}</a> :
            description.text
        }
      </h5>
    }

    {
      content &&
      <span className={`content`}>
        {
          content.map((item) =>
            item.url ?
              <p className={`url`} style={{marginBottom: `5px`}} key={item.text}>
                <a href={item.url}>{item.text}</a>
              </p> :
              <p className={`text`} style={{marginBottom: `5px`}} key={item.text}>{item.text}</p>
          )
        }
      </span>
    }


  </div>

SpeciesCard.propTypes = cardPropTypes

export default SpeciesCard
