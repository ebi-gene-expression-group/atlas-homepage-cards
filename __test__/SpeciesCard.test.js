import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import EbiSpeciesIcon from 'react-ebi-species'

import SpeciesCard from '../src/cards/SpeciesCard'

Enzyme.configure({ adapter: new Adapter() })

describe(`SpeciesCard`, () => {
  test(`with no data matches snapshot`, () => {
    const tree = renderer.create(<SpeciesCard iconSrc={``}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`does not render optional empty content`, () => {
    const wrapper = shallow(<SpeciesCard iconSrc={``} />)

    expect(wrapper.find(EbiSpeciesIcon).exists()).toBe(true)
    expect(wrapper.find(`li`).exists()).toBe(false)
    expect(wrapper.find(`ul`).exists()).toBe(false)
  })

  test(`does not render non-existent URLs`, () => {
    const speciesName = `Mus musculus`
    const content = [
      {
        text: `120 experiments`
      },
      {
        text: `1 baseline experiment`
      },
      {
        text: `119 differential experiments`
      }
    ]

    const description = {
      text: speciesName
    }

    const wrapper = shallow(<SpeciesCard iconSrc={speciesName} content={content} description={description}/>)
    expect(wrapper.find(`h5`).exists()).toBe(true)
    expect(wrapper.find(`h5`).text()).toBe(speciesName)
    expect(wrapper.find(`h5 a`).exists()).toBe(false)

    const contentWrapper = wrapper.find(`ul`)

    expect(contentWrapper.exists()).toBe(true)
    expect(contentWrapper.find(`li`)).toHaveLength(3)
    expect(contentWrapper.find(`li a`).exists()).toBe(false)
  })

  test(`renders URLs`, () => {
    const description = {
      text: `Mus musculus`,
      url: `http://foo.bar/species`
    }

    const content = [
      {
        text: `120 experiments`,
        url: `http://example.com/experiments`
      }
    ]

    const wrapper = shallow(<SpeciesCard iconSrc={``} content={content} description={description}/>)

    expect(wrapper.find(`h5 a`).exists()).toBe(true)
    expect(wrapper.find(`li a`)).toHaveLength(1)
  })
})
