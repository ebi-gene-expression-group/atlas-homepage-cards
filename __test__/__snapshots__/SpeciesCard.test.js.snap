// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeciesCard With no data matches snapshot 1`] = `
<div
  className="column column-block text-center combo card"
  style={
    Object {
      "marginBottom": 0,
      "paddingBottom": "25px",
    }
  }
>
  <span
    className="species-icon"
    style={
      Object {
        "fontSize": "600%",
      }
    }
  >
    <span
      className="react-ebi-species-icon"
      data-icon=""
      style={
        Object {
          "color": undefined,
        }
      }
      title=""
    />
  </span>
</div>
`;
